status.tmpl

```diff
@@ -162,67 +222,108 @@
           }
         }
       };
+
+      var countNodeStatuses = function() {
+        var dbs = {{ marshal . }};
+        var elements = document.getElementsByClassName('versionstate');
+        [].forEach.call(elements, function (elem) {
+          var dbName = elem.getAttribute('data-dbName');
+          var versionName = elem.getAttribute('data-versionName');
+
+          var statusCounts = {};
+          var nodes = dbs.dbs[dbName].versions[versionName].nodes;
+          Object.values(nodes).forEach(function (node) {
+            if (node.state in statusCounts) {
+              statusCounts[node.state]++;
+            } else {
+              statusCounts[node.state] = 1;
+            }
+          });
+
+          for (status in statusCounts) {
+            var count = document.createElement('span');
+            count.innerHTML = statusCounts[status].toString() + 'x';
+            count.classList.add('count');
+            elem.appendChild(count);
+
+            var label = document.createElement('span');
+            label.innerHTML = status;
+            label.classList.add('label');
+            label.classList.add(status);
+            elem.appendChild(label);
+          }
+        });
+      };
     </script>
   </head>
 
-  <body onload="drawMaps();">
+  <body {{ if not (isListView $.DBs) }} onload="drawMaps();" {{ else }} onload="countNodeStatuses();" {{ end }}>
     <div id="header">
       <h1>Sequins!</h1>
     </div>
+
     <div id="wrapper">
       {{ range $dbName, $db := .DBs}}
-      <div class="db">
-        <h2 class="dbname">/{{ $dbName }} {{ if gt (len $.DBs) 1 }}<a href="/{{ $dbName }}">&rarr;</a>{{ end }}</h2>
-        {{ range $versionName, $version := $db.Versions }}
-        <div class="version">
-          <span class="versionname">/{{ $versionName }}</span><span class="versionpath">({{ $version.Path }})</span>
-          <div class="versionwrapper">
-            <div class="nodes">
-              {{ range $nodeName, $node := $version.Nodes }}
-              <div class="node {{ if $node.Current }}nodecurrent{{else}}nodenotcurrent{{end}}{{ if eq $node.State "ERROR" }} nodeerror{{end}}">
-                <div class="nodename">{{$nodeName}}</div>
-                <div class="nodestatus">
-                {{ if eq $node.State "BUILDING" }}
-                  (building since {{ $node.CreatedAt }})
-                {{ else if eq $node.State "ERROR" }}
-                  (errored)
-                {{ else }}
-                  (available since {{ $node.AvailableAt }})
-                {{ end }}
+        <div class="db">
+          <h2 class="dbname">/{{ $dbName }} {{ if (isListView $.DBs) }}<a href="/{{ $dbName }}">&rarr;</a>{{ end }}</h2>
+          {{ range $versionName, $version := $db.Versions }}
+
+          <div class="version">
+            <span class="versionname">/{{ $versionName }}</span>
+
+            {{ if (isListView $.DBs) }}
```

Comment:
@dusty-stripe I added the `isListView` function to the template since I just realized I could :P

It's not any more concise, but hopefully it conveys more meaning?

Reply:
👍